integrations/pgvector/src/haystack_integrations/document_stores/pgvector/document_store.py

```diff
@@ -539,78 +759,63 @@ def write_documents(self, documents: List[Document], policy: DuplicatePolicy = D
         # https://www.psycopg.org/psycopg3/docs/api/cursors.html#psycopg.Cursor.executemany
         written_docs = 0
         while True:
-            if self.cursor.fetchone():
+            if self._cursor.fetchone():
                 written_docs += 1
-            if not self.cursor.nextset():
+            if not self._cursor.nextset():
                 break
 
         return written_docs
 
-    @staticmethod
-    def _from_haystack_to_pg_documents(documents: List[Document]) -> List[Dict[str, Any]]:
+    async def write_documents_async(
+        self, documents: List[Document], policy: DuplicatePolicy = DuplicatePolicy.NONE
+    ) -> int:
         """
-        Internal method to convert a list of Haystack Documents to a list of dictionaries that can be used to insert
-        documents into the PgvectorDocumentStore.
+        Writes documents to the document store asynchronously.
```

Comment:
extend the docstring with input parameters, returns and any errors raised, for example `ValueError` if `documents` is not a `List` of `Document`.